packages/react-email/src/utils/js-email-detection.spec.ts

```diff
@@ -0,0 +1,54 @@
+import fs from 'node:fs';
+import path from 'node:path';
+import { getEmailsDirectoryMetadata } from './get-emails-directory-metadata';
+
+describe('JavaScript Email Detection', () => {
+  const testingDir = path.resolve(__dirname, 'testing');
+
+  test('should detect JavaScript files with ES6 export default syntax', async () => {
+    const emailsMetadata = await getEmailsDirectoryMetadata(testingDir, true);
+
+    expect(emailsMetadata).toBeDefined();
+    expect(emailsMetadata?.emailFilenames).toContain(
+      'js-email-export-default.js',
+    );
+  });
+
+  test('should detect JavaScript files with CommonJS module.exports', async () => {
+    const emailsMetadata = await getEmailsDirectoryMetadata(testingDir, true);
+
+    expect(emailsMetadata).toBeDefined();
+    expect(emailsMetadata?.emailFilenames).toContain('js-email-test.js');
+  });
+
+  test('should detect MDX-style JavaScript files with named exports', async () => {
```

Comment:
use `it` instead of `test` for these kind of descriptions

```suggestion
  it('should detect MDX-style JavaScript files with named exports', async () => {
```